src/lib/stores/walletStore.js

```diff
@@ -95,19 +133,42 @@ function createWalletStore() {
          */
         sign: async ({ transactionXDR, network, pincode }) => {
             try {
-                const keyManager = setupKeyManager()
-                let signedTransaction = await keyManager.signTransaction({
+                const keyManager = setupKeyManager();
+                const { keyId, publicKey } = get(walletStore);
+                
+                if (keyId === publicKey) {
+                    const kit = new StellarWalletsKit({
+                        network: WalletNetwork.TESTNET,
```

Comment:
```suggestion
                        network: network,
```

Reply:
I think this one got missed.

Reply:
Ser for this the network parameter in StellarWalletsKit constructor expects a specific  value (WalletNetwork.TESTNET or WalletNetwork.PUBLIC) that why I didn't change it but I added a tenary operator for both of them

Reply:
You're right that `WalletNetwork.TESTNET` or `WalletNetwork.PUBLIC` will be a specific value, but they will be the same "network passphrase" value that we are passing in as the `network` argument: [see source](https://github.com/Creit-Tech/Stellar-Wallets-Kit/blob/main/src/types.ts)

The "network passphrase" is a bit of a misnomer, but it's used to ensure that all transactions submitted to a given Stellar network are actually _intended_ for that particular network. The values that we may pass into this `sign` function will indeed be the same values held by the `WalletNetwork` object.